_blog/deploying-laravel-to-kubernetes/index.md

```diff
@@ -0,0 +1,289 @@
+
+__Prerequisites__
+
+To follow with this demonstration, you will need the following installed on your local system:
+
+
+1) <a href="https://docs.docker.com/install/" target="_blank" 
+rel="noopener">Docker</a>
+
+2) <a href="https://kubernetes.io/docs/tasks/tools/install-kubectl/" target="_blank" 
+rel="noopener">Kubectl</a>
+
+3) <a href="https://github.com/kubernetes/minikube/releases" target="_blank" 
+   rel="noopener">Minikube</a>
+
+_Are you having problems installing and running these applications on Windows? LearnK8s will be publishing a tutorial on how to accomplish this very soon. Subscribe and be notified as soon as new articles get published._
+
+
+__Docker image__
+
+Kubernetes deploys containerised applications, and therefore as a first step, you will need to build a Docker image of the demo application. Since this tutorial will be run locally on Minikube, you can just build a local Docker Image from the `DockerFile` included in the example code. 
+
+```bash
+FROM php:7
+RUN apt-get update -y && apt-get install -y openssl zip unzip git
+RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
+RUN docker-php-ext-install pdo mbstring
+WORKDIR /app
+COPY . /app
+RUN composer install
+
+RUN php artisan key:generate
+RUN php artisan serve --host=0.0.0.0 --port=8181
+EXPOSE 8181
+``` 
+This `DockerFile` is reasonably basic:
+
+It extends a PHP7 base image, installs some system dependencies including composer and the standard PHP extensions required by Laravel. It then copies the application files to a working directory and installs the application's dependencies via composer. It also runs a couple of `artisan` commands,  last of which serves the application using the built-in PHP web server. In the end, it exposes port `8181` to the host machine.
+  
+To build the local Docker image:
+
+```bash
+cd /to/your/project/directory
+eval $(minikube docker-env)
+docker build -t yourname/laravel-kubernetes-demo .
+```
+
+__Deploying the image__
+
+Now that the application's image is built and available locally you can go ahead with deploying it.
+
+I always start with making sure that Kubectl is in the correct context. In this case, the context is Minikube. You can quickly switch context as follows:
+
+```bash
+kubectl config use-context minikube
+```
+
+then you can deploy the container image:
+
+```bash
+kubectl run laravel-kubernetes-demo --image=yourname/laravel-kubernetes-demo 
+--port=8181 --image-pull-policy=IfNotPresent
+```
+The above command tells Kubectl to run our demo application from the Docker image while making port 8181 available for listening. The last parameter of the command simply asks Kubectl to not pull the image from a registry such as Docker Hub if it exists locally which in this case it does. Do note that you still need to be logged on to Docker's so that Kubectl can check if the image is up to date.
+
+You can check that a Pod is created for our application by running:
```

Comment:
for your application?

Reply:
Should be _the application_. I'll fix it.